docs/guides/scripts.md

```diff
@@ -1 +1,175 @@
-# Running a script
+# Running scripts
+
+## Running a script without dependencies
+
+If a script has no dependencies, it can be executed with `uv run`:
+
+```python
+print("Hello world")
+```
+
+```console
+$ uv run example.py
+Hello world
+```
+
+Similarly, if the script depends on a module in the standard library, there's nothing more to do:
+
+```python
+import os
+
+print(os.path.expanduser("~"))
+```
+
+```console
+$ uv run example.py
+/Users/astral
+```
+
+Arguments can be passed to the script:
+
+```python
+import sys
+
+print(" ".join(sys.argv[1:]))
+```
+
+```console
+$ uv run example.py test
+test
+
+$ uv run example.py hello world!
+hello world!
+```
+
+Note that if `uv run` is used in a _project_, i.e. a directory with a `pyproject.toml`, it will install the current project before running the script. If the script does not depend on the project, use the `--isolated` flag to skip this:
+
+```console
+# Note, it is important that the flag comes _before_ the script
+$ uv run --isolated example.py
+```
+
+See the [projects](./projects.md) guide for more details on working in projects.
+
+## Running a script with dependencies
+
+When a script requires dependencies, they must be installed into the environment that the script runs in. uv prefers to create these environments on-demand instead of having a long-lived virtual environment with manually managed dependencies. This requires declaration
```

Comment:
```suggestion
When a script requires dependencies, they must be installed into the environment that the script runs in. uv prefers to create these environments on-demand instead of maintaining a long-lived virtual environment with manually managed dependencies. This requires explicit declaration
```